autoload/go/config.vim

```diff
@@ -15,10 +15,8 @@ function! go#config#BuildTags() abort
 endfunction
 
 function! go#config#SetBuildTags(value) abort
-  if a:value == ""
-    if exists('g:go_build_tags')
-      unlet g:go_build_tags
-    endif
+  if a:value is '' || a:value is '""' || a:value is "''"
```

Comment:
The conversion from `''` and `""` to the empty string should happen in `go#cmd#BuildTags`. I'll submit a separate PR to do that so it's resolved separately...

Reply:
Alright, I merged your PR and rebased on master.